tests/app/main/views/test_communications.py

```diff
@@ -39,88 +52,198 @@ def test_get_page_should_only_be_accessible_to_specific_user_roles(self, role, e
         assert actual_code == expected_code, "Unexpected response {} for role {}".format(actual_code, role)
 
     def test_page_shows_empty_messages_if_no_files_uploaded(self):
-        self.s3.return_value.list.side_effect = ([], [])  # Empty lists for both communication and clarification files
-        self.data_api_client.get_framework.return_value = {"frameworks": {"status": "open"}}
+        self.s3.return_value.list.side_effect = lambda *args, **kwargs: []  # return a fresh empty list every time
         response = self.client.get("/admin/communications/{}".format(self.framework_slug))
         document = html.fromstring(response.get_data(as_text=True))
-        file_upload_messages = document.xpath('//p[@class="file-upload-existing-value"]')
+        file_upload_messages = document.xpath('//p[@class="summary-item-no-content"]')
 
         assert len(file_upload_messages) == 2
-        assert file_upload_messages[0].text_content().strip() == "No communications have been uploaded yet"
-        assert file_upload_messages[1].text_content().strip() == "No clarification answers have been uploaded yet"
+        assert file_upload_messages[0].text_content().strip() == "No communications files"
+        assert file_upload_messages[1].text_content().strip() == "No clarification answers"
+
+        assert self.data_api_client.mock_calls == [
+            mock.call.get_framework(self.framework_slug)
+        ]
+        assert self.s3.mock_calls == [
+            mock.call("flop-slop-slap"),
+            mock.call().list('g-things-23/communications/updates/communications', load_timestamps=True),
+            mock.call().list('g-things-23/communications/updates/clarifications', load_timestamps=True),
+        ]
```

Comment:
I feel like these assertions would be clearer if we attached the return value of the s3 mock to the test class; then we could write things like `self.communications_bucket.list.mock_calls`

Reply:
I think that would introduce more possibilities for ... weirdness. People thinking you can just switch out the `self.communications_bucket` in a test and it'll start returning that... I'd also need to start adding another line to separately test that the constructor is called with the right parameter. Really, `mock_calls` is very powerful and gives us quite a nice concise way to describe the things that have happened to the object/class and I think we should use it more tbh.